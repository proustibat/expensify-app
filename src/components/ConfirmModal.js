import React from 'react';
import Modal from 'react-modal';

export default class ConfirmModal extends React.Component {

    customStyles = {
        content : {
            top                   : '50%',
            left                  : '50%',
            right                 : 'auto',
            bottom                : 'auto',
            marginRight           : '-50%',
            transform             : 'translate(-50%, -50%)'
        }
    };

    constructor( props ) {
        super( props );

        // Make sure to bind modal to your appElement (http://reactcommunity.org/react-modal/accessibility/)
        Modal.setAppElement( document.querySelector( '#app' ) );
    }

    render() {
        return (
            <Modal
                isOpen = { this.props.showModal }
                contentLabel = { 'Confirm choice modal' }
                style = { this.customStyles}
                onRequestClose = { this.props.handleCloseModal }
            >
                {/*Are you sure you wanna delete this expense?*/}
                <p>{ this.props.body }</p>
                <div className="button-container--space-around">
                    <button className="button button--cancel" onClick={ this.props.handleCloseModal }>{ this.props.cancelText }</button>
                    <button className="button button--valid" onClick={ this.props.handleConfirm }>{ this.props.confirmText }</button>
                </div>
            </Modal>
        );
    }
}

