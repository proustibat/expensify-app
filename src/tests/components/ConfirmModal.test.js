import React from "react";
import Modal from 'react-modal';
import { shallow } from "enzyme";
import ConfirmModal from "../../components/ConfirmModal";


let handleCloseModal, handleConfirm, body, wrapper, confirmText, cancelText;

beforeEach( () => {
    handleCloseModal = jest.fn();
    handleConfirm = jest.fn();
    body = 'my tst body';
    confirmText = 'confirm text';
    cancelText =  'cancel text';
    wrapper = shallow( <ConfirmModal
        showModal = { true }
        handleCloseModal = { handleCloseModal }
        handleConfirm = { handleConfirm }
        body = { body }
        confirmText={ confirmText }
        cancelText={ cancelText }
    /> );
} );

test( 'Should render the confirm modal correctly', () => {
    expect( wrapper ).toMatchSnapshot();
} );

test( 'Should render a react modal', () => {
    expect( wrapper.find( Modal ).length ).toEqual( 1 );
} );

test( 'Should opens modal', () => {
    expect( wrapper.find( Modal ).prop( 'isOpen' ) ).toEqual( true );
} );

test( 'Should handle handleConfirm method when confirming action', () => {
    wrapper.find( '.button--valid' ).simulate( 'click' );
    expect( handleConfirm ).toHaveBeenCalled();
} );

test( 'Should handle handleCloseModal method when canceling action', () => {
    wrapper.find( '.button--cancel' ).simulate( 'click' );
    expect( handleCloseModal ).toHaveBeenCalled();
} );
