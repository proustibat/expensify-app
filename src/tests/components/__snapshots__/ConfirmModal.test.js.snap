// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the confirm modal correctly 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  closeTimeoutMS={0}
  contentLabel="Confirm choice modal"
  isOpen={true}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
  style={
    Object {
      "content": Object {
        "bottom": "auto",
        "left": "50%",
        "marginRight": "-50%",
        "right": "auto",
        "top": "50%",
        "transform": "translate(-50%, -50%)",
      },
    }
  }
>
  <p>
    my tst body
  </p>
  <div
    className="button-container--space-around"
  >
    <button
      className="button button--cancel"
      onClick={[Function]}
    >
      cancel text
    </button>
    <button
      className="button button--valid"
      onClick={[Function]}
    >
      confirm text
    </button>
  </div>
</Modal>
`;
